src/main/java/com/yahoo/sketches/theta/MWMRHeapQuickSelectSketch.java

```diff
@@ -0,0 +1,86 @@
+package com.yahoo.sketches.theta;
+
+import static com.yahoo.sketches.Util.MIN_LG_ARR_LONGS;
+import static com.yahoo.sketches.theta.PreambleUtil.MAX_THETA_LONG_AS_DOUBLE;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import com.yahoo.sketches.Family;
+import com.yahoo.sketches.ResizeFactor;
+import com.yahoo.sketches.Util;
+
+public class MWMRHeapQuickSelectSketch extends HeapQuickSelectSketch {
```

Comment:
Idit suggested that we wouldn't use MWMR but concurrent instead

Reply:
Change to composition instead of inheritance to support concurrency of all hierarchy